dask_cuda/initialize.py

```diff
@@ -70,18 +71,7 @@ def initialize(
                 if net_devices is not None and net_devices != "":
                     options["NET_DEVICES"] = net_devices
 
-            ucp.reset()
-            ucp.init(options=options)
-
-            ucx_env = {}
-            for k, v in ucp.get_config().items():
-                # Skip values that aren't actual environment variables (i.e., not strings)
-                if isinstance(v, str):
-                    ucx_env["UCX_" + k] = v
-
-            # Set also UCX environment variables: required by Dask client. It may be best ti
-            # to have the client asking the scheduler for the proper variables.
-            os.environ.update(ucx_env)
+            dask.config.set({"ucx": options})
```

Comment:
This won't work because the preload script runs _after_ communication is started (see discussion in https://github.com/dask/distributed/pull/3193). That said, we need to make sure we pass the appropriate parameters down to distributed itself (more specifically https://github.com/dask/distributed/blob/master/distributed/comm/ucx.py), to make sure correct options are read by UCX at the very beginning.

Reply:
Using https://github.com/dask/distributed/pull/3178, the options are passed to `ucx.py`.

With this PR https://github.com/dask/distributed/pull/3192, I was hoping we could avoid using the preload scripts